server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1060,21 +1033,19 @@ private static boolean assertNoDanglingSnapshots(ClusterState state) {
             .filter(entry -> entry.state() == SnapshotDeletionsInProgress.State.STARTED)
             .map(SnapshotDeletionsInProgress.Entry::repository)
             .collect(Collectors.toSet());
-        final Set<String> reposSeen = new HashSet<>();
-        for (SnapshotsInProgress.Entry entry : snapshotsInProgress.entries()) {
-            if (reposSeen.add(entry.repository())) {
-                for (ObjectCursor<ShardSnapshotStatus> value : entry.shardsByRepoShardId().values()) {
-                    if (value.value.equals(ShardSnapshotStatus.UNASSIGNED_QUEUED)) {
-                        assert reposWithRunningDelete.contains(entry.repository())
-                            : "Found shard snapshot waiting to be assigned in [" + entry + "] but it is not blocked by any running delete";
-                    } else if (value.value.isActive()) {
-                        assert reposWithRunningDelete.contains(entry.repository()) == false
-                            : "Found shard snapshot actively executing in ["
-                                + entry
-                                + "] when it should be blocked by a running delete ["
-                                + Strings.toString(snapshotDeletionsInProgress)
-                                + "]";
-                    }
+        for (List<SnapshotsInProgress.Entry> repoEntry : snapshotsInProgress.entriesByRepo().values()) {
+            final SnapshotsInProgress.Entry entry = repoEntry.get(0);
```

Comment:
Not directly related to this change, but could/should we check later snapshots too? We'd expect the first mention of every shard in the list to be active (or blocked by a delete) I think.

Reply:
Yea there's a couple of other week points in this assertion that https://github.com/elastic/elasticsearch/pull/78587 revealed in fact. I think I'll be able to go for a set of way cleaner and stricter assertions in the follow-up to this one.